torch/csrc/jit/script/compiler.cpp

```diff
@@ -2420,6 +2229,65 @@ struct to_ir {
     return graph->insertConstant(stack[0], nullptr, tree->range());
   }
 
+  // We construct the iterable tree here using the IterableTree SugaredValue,
+  // The tree consists of SimpleValue, RangeValue or IterableValue:
+  // For SimpleValues(List, Dict, etc) or RangeValue. We will make them as tree leaves
+  // since we could get the loop information from len() and get_item().
+  // For IterableValue like zip(), enumerate(), we can model them as a combination of
+  // leaves, and we emit a IterableTree value to record the tree information
+  SugaredValuePtr emitIterableTree(
+      SourceRange& loc,
+      const List<Expr>& inputs,
+      const std::shared_ptr<IterableValue>& iterable) {
+      std::shared_ptr<IterableTree> iterable_tree = nullptr;
+      size_t input_size = inputs.size();
+
+      // Handling different iterable values
+      if (iterable->symbol_ == prim::range) {
+        std::vector<Value*> input_vals = getValues(inputs, /*maybe_unpack=*/true);
+        return std::make_shared<RangeValue>(loc, method, input_vals);
+      } else if (iterable->symbol_ == prim::enumerate) {
+        // enumerate(x) can be rewrite as subtrees: (range(0, math.inf), x)
```

Comment:
maybe we could expand your comment into `IterableTree(RangeValue(0, math.inf), SimpleValue(x)`